client/resource_source.go

```diff
@@ -770,17 +930,45 @@ func parseFileFormat(fileFormat *FileFormat) map[string]interface{} {
 	fileFormatMap := make(map[string]interface{})
 	fileFormatMap["file_format"] = fileFormat.Type
 	if fileFormat.Type == "csv" {
+		fileFormatMap["compression"] = fileFormat.Compression
+		fileFormatMap["date_format"] = fileFormat.DateFormat
+		fileFormatMap["empty_value"] = fileFormat.EmptyValue
+		fileFormatMap["field_separator"] = fileFormat.Sep
+		fileFormatMap["ignore_leading_whitespace"] = fileFormat.IgnoreLeadingWhiteSpace
+		fileFormatMap["ignore_trailing_whitespace"] = fileFormat.IgnoreTrailingWhiteSpace
 		fileFormatMap["include_header"] = fileFormat.IncludeHeader
+		fileFormatMap["line_separator"] = fileFormat.LineSep
+		fileFormatMap["quote_all"] = fileFormat.QuoteAll
+		fileFormatMap["timestamp_format"] = fileFormat.TimestampFormat
 	}
 	return fileFormatMap
 }
 
 func composeFileFormat(d map[string]interface{}) *FileFormat {
 	if d["file_format"] == "csv" {
+		compression := string(d["compression"].(string))
```

Comment:
Do you need the explicit `string` conversion after the type assertion?

Reply:
Probably not.